backend/msgpack/writer.go

```diff
@@ -142,11 +147,7 @@ func (w *writer) Close() error {
 		return errInstanceWriterClosed
 	}
 	w.closed = true
-	if err := w.flushWithLock(); err != nil {
-		w.log.WithFields(
-			xlog.NewLogErrField(err),
-		).Error("flush error when closing writer")
-	}
+	w.flushWithLock() // nolint: errcheck
```

Comment:
Hm, not emitting the error here? I usually do it at callsite instead of inside the function itself. If you do it this way, you don't need the `// nolint: errcheck`.